build/android/app/src/main/java/net/minetest/minetest/CopyZipTask.java

```diff
@@ -51,11 +52,11 @@ protected void onPostExecute(String result) {
 
 	private void copyAsset(String zipName) {
 		String filename = zipName.substring(zipName.lastIndexOf("/") + 1);
-		try (InputStream in = contextRef.get().getAssets().open(filename);
-		     OutputStream out = new FileOutputStream(zipName)) {
+		try (InputStream in = activityRef.get().getAssets().open(filename);
+			 OutputStream out = new FileOutputStream(zipName)) {
 			copyFile(in, out);
 		} catch (IOException e) {
-			Toast.makeText(contextRef.get(), e.getLocalizedMessage(), Toast.LENGTH_LONG).show();
+			activityRef.get().runOnUiThread(() -> Toast.makeText(activityRef.get(), e.getLocalizedMessage(), Toast.LENGTH_LONG).show());
```

Comment:
should probably check the reference before access

Reply:
Not sure. It is already Weak Reference and will keep activity reference even it is destroyed already.

Reply:
[WeakReference](https://docs.oracle.com/javase/8/docs/api/java/lang/ref/WeakReference.html) will return null if the activity is destroyed

```java
Activity activity = activityRef.get();
if (activity != null) {
	// stuff
}
```

Reply:
Yes, sorry. I totally messed up with references.
Fix pushed. One more improvement added.